builtin/providers/aws/resource_aws_security_group_rule.go

```diff
@@ -334,25 +337,25 @@ func resourceAwsSecurityGroupRuleDelete(d *schema.ResourceData, meta interface{}
 	return nil
 }
 
-func findResourceSecurityGroup(conn *ec2.EC2, id string) (*ec2.SecurityGroup, error) {
-	req := &ec2.DescribeSecurityGroupsInput{
-		GroupIds: []*string{aws.String(id)},
-	}
-	resp, err := conn.DescribeSecurityGroups(req)
-	if err, ok := err.(awserr.Error); ok && err.Code() == "InvalidGroup.NotFound" {
-		return nil, securityGroupNotFound{id, nil}
-	}
+func findResourceSecurityGroup(conn *ec2.EC2, cache *Cache, id string) (*ec2.SecurityGroup, error) {
+	var group *ec2.SecurityGroup
+	err := updateSecurityGroupCache(conn, cache)
+
 	if err != nil {
 		return nil, err
 	}
-	if resp == nil {
-		return nil, securityGroupNotFound{id, nil}
+
+	for _, sg := range cache.SecurityGroups.SecurityGroups {
```

Comment:
Do we need to lock around our access here? Assuming the above `updateCache` method can change this slice?